referenceframe/model.go

```diff
@@ -304,11 +304,12 @@ func ModelFromPath(modelPath, name string) (Model, error) {
 }
 
 // New2DMobileModelFrame builds the kinematic model associated with the kinematicWheeledBase
-// This model is intended to be used with a mobile base and has 3DOF corresponding to a state of x, y, and theta
-// where x and y are the positional coordinates the base is located about and theta is the rotation about the z axis.
+// This model is intended to be used with a mobile base and has either 2DOF corresponding to  a state of x, y
+// or has 3DOF corresponding to a state of x, y, and theta, where x and y are the positional coordinates
+// the base is located about and theta is the rotation about the z axis.
 func New2DMobileModelFrame(name string, limits []Limit, collisionGeometry spatialmath.Geometry) (Model, error) {
-	if len(limits) != 2 {
-		return nil, errors.Errorf("Must have 2DOF state (x, y) to create 2DMobildModelFrame, have %d dof", len(limits))
+	if len(limits) != 2 && len(limits) != 3 {
+		return nil, errors.Errorf("Must have 2DOF state (x, y) to create 2DMobileModelFrame, have %d dof", len(limits))
```

Comment:
Good catch fixing the spelling error! We should also update the rest of the error message to be consistent with the new behavior